contentgrid-spring-data-support/src/test/java/com/contentgrid/data/support/auditing/v1/AuditorAwareImplTest.java

```diff
@@ -0,0 +1,304 @@
+    }
+
+    String formatInstant(Instant date) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH)
+                .withZone(ZoneId.of("GMT"));
+        return formatter.format(date);
+    }
+
+    static JwtRequestPostProcessor jwtWithClaims(String subject, String name) {
+        return jwt().jwt(jwt -> jwt.subject(subject).claim("name", name)
+                .issuer("http://localhost/realms/cg-invalid"));
+    }
+
+    static JwtRequestPostProcessor john = jwtWithClaims("123456", "john smith");
+    static JwtRequestPostProcessor bob = jwtWithClaims("abcdef", "bob admin");
+
+    @Test
+    void postEntity_shouldSetAuditMetadataFields_http201() throws Exception {
+        mockMvc.perform(post("/invoices")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {"number": "123456"}
+                                """)
+                        .with(john))
+                .andExpect(status().isCreated());
+        var dateAfterCreation = Instant.now();
+
+        mockMvc.perform(get("/invoices").with(john))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$._embedded.['d:invoices'][0].number").value("123456"))
+                .andExpect(jsonPath("$._embedded.['d:invoices'][0].auditing").exists())
+                .andExpect(jsonPath("$._embedded.['d:invoices'][0].auditing.created_by").value("john smith"))
+                .andExpect(jsonPath("$._embedded.['d:invoices'][0].auditing.created_date").exists())
+                .andExpect(jsonPath("$._embedded.['d:invoices'][0].auditing.last_modified_by").value("john smith"))
+                .andExpect(jsonPath("$._embedded.['d:invoices'][0].auditing.last_modified_date").exists());
+
+        assertThat(invoices.findAll()).singleElement().satisfies(invoice -> {
+            assertThat(invoice.getNumber()).isEqualTo("123456");
+            assertThat(invoice.getAuditing().getCreatedBy().getId()).isEqualTo("123456");
+            assertThat(invoice.getAuditing().getCreatedDate()).isBefore(dateAfterCreation);
+            assertThat(invoice.getAuditing().getLastModifiedBy().getNamespace()).isEqualTo("http://localhost/realms/cg-invalid");
+            assertThat(invoice.getAuditing().getLastModifiedDate()).isBefore(dateAfterCreation);
+        });
+    }
+
+    @Test
+    void postEntity_withoutJwt_shouldNotCreate_http403() throws Exception {
+        mockMvc.perform(post("/invoices")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {"number": "123456"}
+                                """))
+                .andExpect(status().isForbidden());
+        mockMvc.perform(get("/invoices").with(john))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$._embedded.['d:invoices'].length()").value("0"));
+    }
```

Comment:
Why should this result in a 403?

Allowing or denying access is not the job of the application itself, but of the gateway in front of it.

So, there doesn't have to be a test for that here.